src/Glutinum.Converter/Reader/TypeNode.fs

```diff
@@ -150,6 +152,85 @@ module UtilityType =
+    (reader: ITypeScriptReader)
+    (typeNodeQuery: Ts.TypeQueryNode)
+    =
+    let checker = reader.checker
+    let typ = checker.getTypeAtLocation !!typeNodeQuery.exprName
+
+    resultOption {
+        // This is safe as both cases have a `kind` field
+        let exprNameKind: Ts.SyntaxKind = typeNodeQuery.exprName?kind
+        // This logic is highly overlapping with readTypeUsingFlags
+        // It should be refactored to achieve a sane flow
+        match typ.flags, typ.getSymbol (), exprNameKind with
+        | HasTypeFlags Ts.TypeFlags.Object, None, Ts.SyntaxKind.Identifier ->
+            let exprName: Ts.Identifier = !!typeNodeQuery.exprName
+
+            let! aliasSymbol =
+                checker.getSymbolAtLocation exprName
+                |> ResultOption.ofOption
+                |> Result.mapError (fun _ ->
+                    generateReaderError
+                        "type node (TypeQuery)"
+                        "Missing symbol"
+                        typeNodeQuery
+                )
+
+            let! declaration =
+                aliasSymbol.declarations
+                |> Option.map (fun declarations ->
+                    if declarations.Count <> 1 then
+                        None
+                    else
+                        Some declarations.[0]
+                )
+                |> ResultOption.ofOption
+                |> Result.mapError (fun _ ->
+                    generateReaderError
+                        "type node (TypeQuery)"
+                        "Expected exactly one declaration"
+                        typeNodeQuery
+                )
+
+            let! variableDeclaration =
+                declaration
+                |> Option.bind (fun declaration ->
+                    match declaration.kind with
+                    | Ts.SyntaxKind.VariableDeclaration ->
+                        Some(declaration :?> Ts.VariableDeclaration)
+                    | _ -> None
+                )
+                |> ResultOption.ofOption
+                |> Result.mapError (fun _ ->
+                    generateReaderError
+                        "type node (TypeQuery)"
+                        "Unsupported declaration kind"
+                        typeNodeQuery
+                )
+
+            let! typeNode =
+                variableDeclaration.``type``
+                |> ResultOption.ofOption
+                |> Result.mapError (fun _ ->
+                    generateReaderError
+                        "type node (TypeQuery)"
+                        "Missing type"
+                        typeNodeQuery
+                )
+
+            match typeNode.kind with
+            | Ts.SyntaxKind.TypeOperator ->
+                let typeOperatorNode = typeNode :?> Ts.TypeOperatorNode
+
+                return reader.ReadTypeOperatorNode typeOperatorNode
+
+            | _ -> return readTypeUsingFlags reader typ
+
+        | _ -> return readTypeUsingFlags reader typ
+    }
```

Comment:
@roboz0r 

I gave another chance to rewrite the `TypeQuery` code. Do you think this version of the code is easier to read?

Personally, I think it is easier to read because it remove (at least visually) several level of pattern matching over options.

Reply:
Happy to use the CE to flatten the nested pattern magic but I'd really like to be able to combine and refactor this function and `readTypeUsingFlags` to avoid the overlapping logical branches. I couldn't work out an easy way to do it yesterday but if you think it's a good idea too, I'll give it another go. 

Maybe it should also be split off into a separate `Reader/TypeQueryNode.fs` file

Reply:
> Happy to use the CE to flatten the nested pattern matching but I'd really like to be able to combine and refactor this function and `readTypeUsingFlags` to avoid the overlapping logical branches. I couldn't work out an easy way to do it yesterday but if you think it's a good idea too, I'll give it another go.

I didn't look too much on that, but if you can refactor the code to have a single function/path I am all for it. With or without the CE depending on what works best.

Reply:
@MangelMaxime  I came up with an alternative rewrite of this code here https://github.com/glutinum-org/cli/commit/031dcd5a0131c9184a5a243ffa7d494f4d0a646c I found that just `result {}` rather than `resultOption` was sufficient.

Reply:
I left a few comments on your commit but I think we are on the good track. If we fix the comments, we should be able to merge it in Glutinum and keep moving forward.